### PR TITLE
fix: fss conformity custom check bucketname / path generation

### PR DESCRIPTION
**Pull request by TomRyan-321**
# Title 
Fix bucket & fileprefix object when bucket is in region other than us-east-1

## Change Summary
Fixes the split to support regions other than us-east-1
## PR Checklist

- [X] I've read and followed the [Contributing Guide](https://github.com/trendmicro/cloudone-filestorage-plugins/blob/master/.github/CONTRIBUTING.md).
- Documents/Readmes
    - [ ] Updated accordingly
    - [X] Not required

## Other Notes

Error without this fix:
`[ERROR] IndexError: list index out of range
Traceback (most recent call last):
  File "/var/task/handler.py", line 56, in lambda_handler
    fileprefix = message["file_url"].split(".s3.amazonaws.com/")[1]
[ERROR] IndexError: list index out of range Traceback (most recent call last):   File "/var/task/handler.py", line 56, in lambda_handler     fileprefix = message["file_url"].split(".s3.amazonaws.com/")[1]`

### CODE REVIEW

File: post-scan-actions/aws-python-conformity-custom-check/handler.py [hunk 1]
```diff
@@ -52,8 +52,8 @@ def lambda_handler(event, context):
             ccaccountid = get_cc_accountid(account_id)
 
             # Get the bucket & object details
-            bucketname = message["file_url"].split(".s3.amazonaws.com/")[0][8:]
-            fileprefix = message["file_url"].split(".s3.amazonaws.com/")[1]
+            bucketname = message["file_url"].split(".s3.")[0][8:]
+            fileprefix = message["file_url"].split(".amazonaws.com/")[1]
             filename = message["file_url"].split("/")[-1]
```
Comment (carpusherw) [inline post-scan-actions/aws-python-conformity-custom-check/handler.py:57]: Consider what we've been using in post-scan-action-tag.
```javascript
import urllib.parse

s3_domain_pattern = 's3(\..+)?\.amazonaws.com'
url = urllib.parse.urlparse(message['file_url'])
# check pre-signed URL type, path or virtual
if re.fullmatch(s3_domain_pattern, url.netloc):
    bucket = url.path.split('/')[1]
    s3_object = '/'.join(url.path.split('/')[2:])
else:
    bucket = url.netloc.split('.')[0]
    s3_object = url.path[1:]

object_key = urllib.parse.unquote_plus(s3_object)
```

**Review by carpusherw — CHANGES_REQUESTED** [review]

File: post-scan-actions/aws-python-conformity-custom-check/handler.py (hunk 1 above)
Comment (carpusherw) [reply post-scan-actions/aws-python-conformity-custom-check/handler.py:57]: or https://github.com/trendmicro/cloudone-filestorage-plugins/blob/master/post-scan-actions/aws-python-promote-or-quarantine/handler.py#L50-62

Comment (TomRyan-321) [reply post-scan-actions/aws-python-conformity-custom-check/handler.py:57]: Updating now, will push changes through later tonight

File: post-scan-actions/aws-python-conformity-custom-check/handler.py [hunk 2]
```diff
@@ -52,9 +55,16 @@ def lambda_handler(event, context):
...
+            s3_domain_pattern = 's3(\..+)?\.amazonaws.com'
+            url = urllib.parse.urlparse(message['file_url'])
+            # check pre-signed URL type, path or virtual
+            if re.fullmatch(s3_domain_pattern, url.netloc):
+                bucketname = url.path.split('/')[1]
+                fileprefix = '/'.join(url.path.split('/')[2:])
+            else:
+                bucketname = url.netloc.split('.')[0]
+                fileprefix = url.path[1:]
+            filename = fileprefix.split('/')[-1]
```
Comment (carpusherw) [inline post-scan-actions/aws-python-conformity-custom-check/handler.py:67]: You missed the `urllib.parse.unquote_plus`

File: post-scan-actions/aws-python-conformity-custom-check/handler.py [hunk 3]
```diff
@@ -1,8 +1,11 @@
 import boto3
 import json
 import os
+import re
+import urllib.parse
 import urllib3
 
+
```
Comment (carpusherw) [inline post-scan-actions/aws-python-conformity-custom-check/handler.py:8]: Could remove the extra blank line.

**Comment by TomRyan-321** [comment]
Hopefully all cleared up now, let me know if there's nothing else and I'll squash the commits tomorrow.

**Review by carpusherw — COMMENTED** [review]
LGTM. Please proceed to squash the commits.

**Comment by TomRyan-321** [comment]
Happy Friday! All squashed :)

**Review by carpusherw — APPROVED** [review]
Review Approved

**Comment by carpusherw** [comment]
@chenjill or @jack-c-tang 
May I have an ack to publish the fix of the plugin?

**Comment by chenjill** [comment]
ack

**Pull request merged**